dts/dts.cmake

```diff
@@ -20,21 +20,29 @@ if(CONFIG_HAS_DTS)
     -include ${DTS_SOURCE}
     )
 
-  if(NOT DTC_OVERLAY_FILE)
-    # overlay file is not set, so try to use overlay dir instead.
+  set_ifndef(BOARD_OVERLAY_DIR ${APPLICATION_SOURCE_DIR})
 
-    set_ifndef(DTC_OVERLAY_DIR ${APPLICATION_SOURCE_DIR})
-
-    set(DTC_OVERLAY_FILE ${DTC_OVERLAY_DIR}/${BOARD_FAMILY}.overlay)
+  set(BOARD_OVERLAY_FILE ${DTC_OVERLAY_DIR}/${BOARD_FAMILY}.overlay)
+  if(NOT EXISTS ${BOARD_OVERLAY_FILE})
+    unset(BOARD_OVERLAY_FILE)
   endif()
 
-  if(EXISTS ${DTC_OVERLAY_FILE})
-    set(DTC_INCLUDE_FLAG_FOR_OVERLAY
-      -include ${DTC_OVERLAY_FILE}
-      )
-  else()
-    unset(DTC_INCLUDE_FLAG_FOR_OVERLAY)
-  endif()
+  # Bring in extra overlay files dropped in by the user or anyone else;
+  # make sure they are set at the end so we can override any other setting
+  file(GLOB overlay_files ${APPLICATION_BINARY_DIR}/*.overlay)
```

Comment:
As was gone into depth here: https://github.com/zephyrproject-rtos/zephyr/pull/4677#issuecomment-341362959

we want to avoid using glob from CMake when we can. Unless there
exists a very good use-case for it I think we should diverge from .conf here.

IIRC it was accepted to glob for .conf files because of backwards-compatibility with KBuild, dts does not have this backwards-compatibility requirement luckily so we have the opportunity to achieve a better solution with dts.

Reply:
But if I remove the globbing then we need to settle on a single `.dts` file in the build folder. I have nothing against that, I guess it would be `<build folder>/dts.overlay`. Is that what you suggest?

Reply:
When it comes to configuration files, we let CONF_FILE be a list.

Could we do the same thing with DTC_OVERLAY_FILE?

Perhaps then the MCUboot "magic" could just append some absolute paths (perhaps keyed on SoC or board name) onto DTC_OVERLAY_FILE. Would that work for your plans, @carlescufi and @SebastianBoe ?

Reply:
Yes, why not. In fact, the current implementation should already work if  `DTC_OVERLAY_FILE` is a list, so I can remove the drop-in there. Nice suggestion!